UIKIT_DZ_Clone_AppStore/UIKIT_DZ_Clone_AppStore/ViewControllers/GoodsForYouViewController/GoodsForYouViewController.swift

```diff
@@ -9,4 +9,198 @@ import UIKit
+
+        createMainViewLabels()
+        createNewsImageView()
+
+    }
+
+    // MARK: - IBAction
+    @objc private func changeAvatarImage() {
+        let imagePicker = UIImagePickerController()
+        imagePicker.sourceType = .photoLibrary
+        imagePicker.allowsEditing = true
+        imagePicker.delegate = self
+        present(imagePicker, animated: true)
+    }
+
+    // MARK: - Private Methods
+    private func createProductProgressView() {
+        let progresView = UIProgressView()
+        progresView.frame = CGRect(x: 10, y: 125, width: tovarView.frame.width - 20, height: 20)
+        progresView.progressViewStyle = .default
+        progresView.progress = 0.5
+        progresView.progressTintColor = .green
+        progresView.transform.scaledBy(x: 1, y: 2)
+        tovarView.addSubview(progresView)
+    }
+
+    private func createProductView() {
+        tovarView.frame = CGRect(x: 20, y: 100, width: 350, height: 170)
+        tovarView.layer.cornerRadius = 10
+
+        tovarView.layer.shadowColor = UIColor.gray.cgColor
+        tovarView.layer.shadowOpacity = 0.4
+        tovarView.layer.shadowOffset = .zero
+        tovarView.layer.shadowRadius = 6
+
+        tovarView.backgroundColor = .white
+
+        view.addSubview(tovarView)
+
+        createProductImageView()
+        createline(view: tovarView, rect: CGRect(x: 0, y: 110, width: tovarView.frame.width, height: 1))
+    }
+
+    private func createNewsImageView() {
+
+        let newsView = UIImageView()
+        newsView.contentMode = .scaleToFill
+        newsView.frame = CGRect(x: 35, y: 400, width: 40, height: 40)
+        newsView.image = UIImage(named: "square")
+        view.addSubview(newsView)
+    }
+
+    private func createProductImageView() {
+        let imageView = UIImageView()
+        imageView.frame = CGRect(x: 20, y: 20, width: 70, height: 70)
+        imageView.contentMode = .scaleToFill
+        imageView.image = UIImage(named: "apple-airpod")
+        tovarView.addSubview(imageView)
+    }
+
+    private func setNavigationBar() {
+        navigationItem.title = "Для вас"
+        navigationController?.navigationBar.prefersLargeTitles = true
+        let rightBarButton = UIBarButtonItem(customView: Barbutton)
+        navigationItem.rightBarButtonItem = rightBarButton
+    }
+
+    private func createCustomBarButton() {
+
+        let image  = getCustomBarButtonImage()
+        Barbutton.frame = CGRect(x: 0, y: 0, width: 40, height: 40)
+        Barbutton.setImage(image, for: .normal)
+        Barbutton.layer.cornerRadius = 20
+        Barbutton.clipsToBounds = true
+        Barbutton.addTarget(self, action: #selector(changeAvatarImage), for: .touchUpInside)
+    }
+
+    private func getCustomBarButtonImage() -> UIImage {
+
+        let userImage = UserDefaults.standard.value(forKey: "avatar")
+
+        var image = UIImage()
+
+        if let userimage = userImage as? Data {
+            image = UIImage(data: userimage)!
```

Comment:
```suggestion
            image = UIImage(data: userimage)
```
! - следующий ревью  1 ! это 10 приседов